miio/gateway.py

```
@@ -536,38 +513,216 @@ class SubDevice:
+        except Exception as ex:
+            _LOGGER.debug(
+                "Got an exception while fetching properties %s: %s", properties, ex, exc_info=True
+            )
+            return None
+
+    @command(click.argument("property"), click.argument("value"))
+    def set_subdevice_prop(self, property, value):
+        """Set a device property of the subdevice."""
+        try:
+            return self.gw.send("set_device_prop", {"sid": self.sid, property: value})
+        except Exception as ex:
+            _LOGGER.debug(
+                "Got an exception while setting propertie %s to value %s: %s", property, str(value), ex, exc_info=True
+            )
+            return None
+
+    @command()
+    def unpair(self):
+        return self.subdevice_send("remove_device")
+
+    @command()
+    def get_battery(self):
+        self._battery = self.subdevice_send("get_battery").pop()
+        return self._battery
+
+    @command()
+    def get_firmware_version(self) -> Optional[int]:
+        """Returns firmware version"""
+        return self.get_subdevice_prop("fw_ver").pop()
+
+
+class AqaraHT(SubDevice):
+    _temperature = None
+    _humidity = None
+    _pressure = None
+
+    @property
+    def temperature(self):
+        """Return the temperature in degrees celsius"""
+        return self._temperature
+
+    @property
+    def humidity(self):
+        """Return the humidity in %"""
+        return self._humidity
+
+    @property
+    def pressure(self):
+        """Return the pressure in hPa"""
+        return self._pressure
+
+    @command()
+    def update(self):
+        """Update all device properties"""
+        values = self.get_subdevice_prop_exp(["temperature", "humidity", "pressure"])
+        if len(values) == 3:
+            self._temperature = values[0]/100
+            self._humidity = values[1]/100
+            self._pressure = values[2]/100
+        return
```

No need for explicit returns. Error cases should be raised as exceptions instead of silently failing.

---

done